lms/djangoapps/teams/static/teams/js/views/team_profile.js

```diff
@@ -38,7 +38,20 @@
                 render: function() {
                     var memberships = this.model.get('membership'),
                         discussionTopicID = this.model.get('discussion_topic_id'),
-                        isMember = TeamUtils.isUserMemberOfTeam(memberships, this.context.userInfo.username);
+                        isMember = TeamUtils.isUserMemberOfTeam(memberships, this.context.userInfo.username),
+                        isInstructorManagedTopic = false,
+                        isAdminOrStaff = this.context.userInfo.privileged || this.context.userInfo.staff;
+
+                    var topic_id = this.model.get('topic_id');
+                    var topics = this.context.topics.results;
```

Comment:
You can avoid doing this topic-finding loop logic by passing the topic in the options to the [`TeamProfileView` constructor call](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/teams/static/teams/js/views/teams_tab.js#L420-L426), just like the call to `TeamProfileHeaderActionsView` below it does, and then setting `this.topic = options.topic` in the [TeamProfile `initialize` function](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/teams/static/teams/js/views/team_profile.js#L27).